addons/mail/static/src/core/common/suggestion_service.js

```diff
@@ -166,11 +166,28 @@ export class SuggestionService {
                 suggestions.push(partner);
             }
         }
+        const specialSuggestions = this.store.specialMentions
+            .filter(
+                (special) =>
+                    thread &&
+                    special.channel_types.includes(thread.channel_type) &&
+                    cleanedSearchTerm.length >= Math.min(4, special.label.length) &&
+                    (special.label.startsWith(cleanedSearchTerm) ||
+                        cleanTerm(special.description.toString()).includes(cleanedSearchTerm))
+            )
+            .map((special) => ({
+                isSpecial: true,
+                label: special.label,
+                type: special.label,
+                name: special.displayName,
+                description: special.description,
+            }));
         return {
             type: "Partner",
             suggestions: sort
                 ? [...this.sortPartnerSuggestions(suggestions, cleanedSearchTerm, thread)]
                 : suggestions,
+            specialSuggestions,
```

Comment:
As said in previous comment, prefer not complexify the return value of this function, you can put it in `suggestions` too.